docs/spire.md

```diff
@@ -0,0 +1,285 @@
+ │  ready    │     ┌───────────┐                        │        │ SVIDs
+ │           └────►│  TaskRun  ├────────────────────────┘        │
+ │                 │  Pod      │                                 │
+ │                 └───────────┘     TaskRun Entrypointer        │
+ │                   ▲               Sign Result and update      │
+ │ Get               │ Get SVID      TaskRun status with         │
+ │ SPIRE             │               signature + cert            │
+ │ server            │                                           │
+ │ Credentials       │                                           ▼
+┌┴───────────────────┴─────────────────────────────────────────────────────┐
+│                                                                          │
+│   SPIRE Agent    ( Runs as   )                                           │
+│   + CSI Driver   ( Daemonset )                                           │
+│                                                                          │
+└──────────────────────────────────────────────────────────────────────────┘
+```
+
+Initial Setup:
+1. As part of the SPIRE deployment, the SPIRE server attests the agents running on each node in the cluster.
+2. The Tekton Controller is configured to have workload identity entry creation permissions to the SPIRE server.
+3. As part of the Tekton Controller operations, the Tekton Controller will retrieve an identity that it can use to talk to the SPIRE server to register TaskRun workloads.
+
+When a TaskRun is created:
+1. The Tekton Controller creates a TaskRun pod and its associated resources
+1. When the TaskRun pod is ready, the Tekton Controller registers an identity with the information of the pod to the SPIRE server. This will tell the SPIRE server the identity of the TaskRun to use as well as how to attest the workload/pod.
+1. After the TaskRun steps complete, as part of the entrypointer code, it requests an SVID from SPIFFE workload API (via the SPIRE agent socket)
+1. The SPIRE agent will attest the workload and request an SVID.
+1. The entrypointer receives an x509 SVID, containing the x509 certificate and associated private key. 
+1. The entrypointer signs the results of the TaskRun and emits the signatures and x509 certificate to the TaskRun results for later verification.
+
+## Enabling TaskRun result attestations
+
+To enable TaskRun attestations:
+1. Make sure `enable-spire` is set to `"true"` in the `feature-flags` configmap, see [`install.md`](./install.md#customizing-the-pipelines-controller-behavior) for details
+1. Create a SPIRE deployment containing a SPIRE server, SPIRE agents and the SPIRE CSI driver, for convenience, [this sample single cluster deployment](https://github.com/spiffe/spiffe-csi/tree/main/example/config) can be used.
+1. Register the SPIRE workload entry for Tekton with the "Admin" flag, which will allow the Tekton controller to communicate with the SPIRE server to manage the TaskRun identities dynamically.
```

Comment:
I'm not familiar with SPIRE enough to judge here, but does "admin" comply with the principle of least privilege?

Reply:
The admin flag is needed for the Tekton controller to allow for dynamic registration of the taskruns. The spire server itself should be running in another more secure k8s cluster with limited access. Thus, only the registered tekton controller is allowed to dynamically register taskruns. @lumjjb any other thoughts?